sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java

```diff
@@ -0,0 +1,571 @@
+		if(hmiLevel != HMILevel.HMI_FULL){
+			Log.e(TAG, "Cannot start video service if HMILevel is not FULL.");
+			return null;
+		}
+
+		IVideoStreamListener encoderListener = startVideoStream(new VideoStreamingParameters(), isEncrypted);
+		if (encoderListener == null) {
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+			Log.e(TAG, "Cannot create encoderListener");
+			return null;
+		}
+
+		mSdlEncoder = new SdlEncoder();
+		mSdlEncoder.setFrameRate(frameRate);
+		mSdlEncoder.setFrameInterval(iFrameInterval);
+		mSdlEncoder.setFrameWidth(width);
+		mSdlEncoder.setFrameHeight(height);
+		mSdlEncoder.setBitrate(bitrate);
+		mSdlEncoder.setOutputListener(encoderListener);
+		Surface surface = mSdlEncoder.prepareEncoder();
+
+		if(surface != null){
+			stateMachine.transitionToState(StreamingStateMachine.READY);
+		}else{
+			Log.e(TAG, "Cannot create surface.");
+			stateMachine.transitionToState(StreamingStateMachine.ERROR);
+		}
+		return surface;
+	}
+
+	/**
+	 * Starts the MediaCodec encoder utilized in conjunction with the Surface returned via the createOpenGLInputSurface method
+	 * @see #createOpenGLInputSurface(int, int, int, int, int, boolean)
+	 */
+	public void startEncoder(){
+		if(mSdlEncoder != null){
+			mSdlEncoder.startEncoder();
+			stateMachine.transitionToState(StreamingStateMachine.STARTED);
+		}
+	}
+
+	/**
+	 * Releases the MediaCodec encoder utilized in conjunction with the Surface returned via the createOpenGLInputSurface method
+	 * @see #createOpenGLInputSurface(int, int, int, int, int, boolean)
+	 */
+	public void releaseEncoder(){
+		if(mSdlEncoder != null){
+			mSdlEncoder.releaseEncoder();
+			stateMachine.transitionToState(StreamingStateMachine.STOPPED);
+		}
+	}
+
+	/**
+	 * Drains the MediaCodec encoder utilized in conjunction with the Surface returned via the createOpenGLInputSurface method
+	 * @param endOfStream indicates if this is the end of stream
+	 * @see #createOpenGLInputSurface(int, int, int, int, int, boolean)
+	 */
+	public void drainEncoder(boolean endOfStream){
+		if(mSdlEncoder != null) {
+			mSdlEncoder.drainEncoder(endOfStream);
+		}
+	}
+
+	/**
+	 * Opens a video service (service type 11) and subsequently provides an IVideoStreamListener
+	 * to the app to send video data. The supplied VideoStreamingParameters will be set as desired paramaters
+	 * that will be used to negotiate
+	 *
+	 * @param parameters  Video streaming parameters including: codec which will be used for streaming (currently, only
+	 *                    VideoStreamingCodec.H264 is accepted), height and width of the video in pixels.
+	 * @param encrypted Specify true if packets on this service have to be encrypted
+	 *
+	 * @return IVideoStreamListener interface if service is opened successfully and streaming is
+	 *         started, null otherwise
+	 */
+	public IVideoStreamListener startVideoStream(VideoStreamingParameters parameters, boolean encrypted){
```

Comment:
Is there a reason this is `public`? I believe this is better `protected` at least.

Reply:
The naming between `startVideoStream` and `startVideoStreaming` is very confusing. We should provide clearer naming conventions here.

Reply:
Will change to `protected`, make the other method `startRemoteDisplayStreaming`

Reply:
I believe `startVideoStream` is more related to starting the video service so I think maybe it's better named according to that. `startVideoService`. Correct me if I'm wrong though.